src/client/Microsoft.Identity.Client/Cache/CacheFallbackOperations.cs

```diff
@@ -318,8 +318,15 @@ internal static class CacheFallbackOperations
             // if we have more than 1 RT per env, there is smth wrong with the ADAL cache
             if (rtGroupsByEnv.Any(g => g.Count() > 1))
             {
+                //Due to the fact that there is a problem with the ADAL cache that causes this exception, The ADAL cache will be removed when 
```

Comment:
Try to write a unit test for this, there are already a bunch of tests here: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/f032a4e3c6dde02f0c282a7e560ba49657a2f696/tests/Microsoft.Identity.Test.Unit.net45/CacheTests/CacheFallbackOperationsTests.cs#L104